executor/explain.go

```diff
@@ -73,17 +74,22 @@ func (e *ExplainExec) Next(ctx context.Context, req *chunk.Chunk) error {
 func (e *ExplainExec) generateExplainInfo(ctx context.Context) ([][]string, error) {
 	if e.analyzeExec != nil {
 		chk := newFirstChunk(e.analyzeExec)
+		var nextErr, closeErr error
```

Comment:
Suggest,

```go
func (e *EExplainExec) generateExplainInfo(ctx context.Context) (result [][]string, err error) {
...

var nextErr, closeErr error
defer func() 
    closeErr = e.analyzeExec.Close()
    	if nextErr != nil {
			if closeErr != nil {
				err = errors.New(nextErr.Error() + ", " + closeErr.Error())
			} else {
				err =nextErr
			}
		} else {
                err = closeErr
        }
}
...    
}
```

Reply:
If next() or close() encounter errors, this function should return, rather than execute the code after Line:95. This logic is changed if taking the code above.

Reply:
add defer to handle panic